File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Administration/Language/LanguageGrid.cs

namespace BasicApplication.Administration
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [ColumnsKey("Administration.Language"), IdProperty("Id"), NameProperty("LanguageName")]
    [DialogType(typeof(LanguageDialog)), LocalTextPrefix("Administration.Language"), Service("Administration/Language")]
    public class LanguageGrid : EntityGrid<LanguageRow>, IAsyncInit
    {
        public LanguageGrid(jQueryObject container)
            : base(container)
        {
        }
    }
}

File: serene/src/Serene.Web/Modules/Administration/Language/LanguageGrid.ts
import { EntityGrid } from "@serenity-is/corelib";
import { LanguageColumns, LanguageRow, LanguageService } from "../../ServerTypes/Administration";
import { nsAdministration } from "../../ServerTypes/Namespaces";
import { LanguageDialog } from "./LanguageDialog";

export class LanguageGrid extends EntityGrid<LanguageRow, any> {
    static override[Symbol.typeInfo] = this.registerClass(nsAdministration);

    protected override useAsync() { return true; }
    protected override getColumnsKey() { return LanguageColumns.columnsKey; }
    protected override getDialogType() { return LanguageDialog; }
    protected override getIdProperty() { return LanguageRow.idProperty; }
    protected override getLocalTextPrefix() { return LanguageRow.localTextPrefix; }
    protected override getService() { return LanguageService.baseUrl; }

    protected override afterInit() {
        super.afterInit();
    }
    protected override getDefaultSortBy() {
        return [LanguageRow.Fields.LanguageName];
    }
}